apps/dashboard/app/javascript/packs/files/SweetAlert.js

```diff
@@ -0,0 +1,85 @@
+import Swal from 'sweetalert2'
+
+let sweetAlert = null;
+
+$(document).ready(function(){
+  sweetAlert = new SweetAlert();
+  $("#directory-contents").on("swalShowError", function(e,options) {
+    sweetAlert.alertError(options.title, options.message);
+  });
+
+  $("#directory-contents").on("swalShowPrompt", function(e,options) {
+    sweetAlert.alertError(options.title, options.message);
+  });
+
+  $("#directory-contents").on("swalShowInput", function(e,options) {
+    sweetAlert.input(options);
+  });
+
+  $("#directory-contents").on("swalShowLoading", function(e,options) {
+    sweetAlert.loading(options.message);
+  });
+
+  $("#directory-contents").on("swalClose", function() {
+    sweetAlert.close();
+  });
+
+});
+
+class SweetAlert {
+  _swal = null;
+
+  constructor() {
+    this._swal = Swal;

```

Comment:
Similar to another comment. What's the value add here? why not just use `Swal.fire` or `Swal.mixIn`?

Reply:
Trying to minimize Globally scoped variables.

Reply:
Removed private property of _swal from class and am now using Swal from import.